editor/editor_file_system.cpp

```diff
@@ -961,6 +961,14 @@ void EditorFileSystem::_scan_new_dir(EditorFileSystemDirectory *p_dir, Ref<DirAc
 
 		if (fi->uid != ResourceUID::INVALID_ID) {
 			if (ResourceUID::get_singleton()->has_id(fi->uid)) {
+				// Restrict UID dupe warning to first-scan since we know there are no file moves going on yet.
+				if (first_scan) {
+					// Warn if we detect files with duplicate UIDs.
+					String other_path = ResourceUID::get_singleton()->get_id_path(fi->uid);
+					if (other_path != path) {
+						WARN_PRINT_ONCE(vformat("UID duplicate detected between %s and %s.", path, other_path));
```

Comment:
The effect of `WARN_PRINT_ONCE` is that only the first uid collision will be reported, _globally_. You may want to have a `HashSet` of uids already reported if you want to avoid duplicate warnings.

Reply:
Ah ok, I was assuming it would print once per unique message (like the log rollup in VSCode). I think probably it should just be changed to WARN_PRINT.